AlphaWallet/Extensions/UIButton.swift

```diff
@@ -13,4 +13,17 @@ extension UIButton {
 
         setBackgroundImage(colorImage, for: forState)
     }
+    func setBackgroundColor(lightColor: UIColor, darkColor: UIColor, for state: UIControl.State) {
```

Comment:
i am not quite familiar with dark/light modes, but why just not to use `Appearences` for colors, it should apply right color automatically

Reply:
<img width="1840" alt="Screenshot 2022-06-08 at 16 50 54" src="https://user-images.githubusercontent.com/96283950/172634792-0c8bd45f-3585-4d79-bd0b-680eba16ef41.png">

Reply:
Can this be applied to a uibutton in the various states?

Reply:
@eviltofu In case you wane me to respond, can you elaborate? I don't understand.

@oa-s if you have any idea.

Reply:
So the reason why I don't use appearances directly is because sometimes there are errors in the colour definition. I've seen Dusty defined in RGB as 153,153,153 and 151,151,151. In order to correct this, I would have to go through all the colours that use Dusty and correct them manually. 

If you drag and drop colours in the xcassets file, the colours set are copies of the RGB values. If I drag and drop Red to my new color, I will get RGB 255, 0, 0. If I change Red to RGB 250, 0, 0 this change is not reflected in my new colour. 

UIButtons have different states (disabled, enabled, highlighted/pressed, etc). Each of these states have a different colour associated with them. The UIView's backgroundColor attribute won't work. There are also no function calls to set the colour of UIButtons in the various states in iOS 13 (there is one in iOS 15). The ones we have are self written by ourselves by creating and setting background images for UIButtons. The current ones only take in one colour and do not reflect dark/light modes.  So I've created a extension function in UIButton to do that.

Reply:
> So the reason why I don't use appearances directly is because sometimes there are errors in the colour definition. I've seen Dusty defined in RGB as 153,153,153 and 151,151,151. In order to correct this, I would have to go through all the colours that use Dusty and correct them manually.

Use `153` for all Dusty, until we are told otherwise. Should be an easy change. And no, it will have to be 1 value, not 2

> If you drag and drop colours in the xcassets file, the colours set are copies of the RGB values. If I drag and drop Red to my new color, I will get RGB 255, 0, 0. If I change Red to RGB 250, 0, 0 this change is not reflected in my new colour.

Is this about the question you asked earlier where you want to have references created within the asset catalog pointing to another color in the asset catalog? If so, why does it matter here?

> UIButtons have different states (disabled, enabled, highlighted/pressed, etc). Each of these states have a different colour associated with them. The UIView's backgroundColor attribute won't work. There are also no function calls to set the colour of UIButtons in the various states in iOS 13 (there is one in iOS 15). The ones we have are self written by ourselves by creating and setting background images for UIButtons. The current ones only take in one colour and do not reflect dark/light modes. So I've created a extension function in UIButton to do that.

I understand there are functions that take in colors to create images. So far so good.

> The current ones only take in one colour and do not reflect dark/light modes.

But I don't follow this. It only takes in 1 color, but that color should be the light mode version if it's currently not in dark mode, and the dark mode version when it's in dark mode. It doesn't need to store both light and dark mode versions at runtime. Or is there a case where you think so? Can you write a short snippet and illustrate the problem? As short as possible with as short comments as possible?

Reply:
So that the OS will automatically switch the colour for us. Instead of writing software hooks to change the colours ourselves.

Reply:
Summarizing a chat elsewhere for why it accepts lightColor and darkColor (instead of the current color):

```
func setBackgroundColor(lightColor: UIColor, darkColor: UIColor, for state: UIControl.State) {
```

The gist of it is this function builds a (actually 2) background image and this background image has to switch when iOS tells it to, to match light or dark mode (like how colors would behave)

And this is a special one-off case.